### PR TITLE
Update version to 1.0.0

### PR DESCRIPTION
**Pull request by mykhailo-kuchma**
Update version to 1.0.0, Update CHANGELOG.md

Fix typo in README about CMake minimal version.

Resolves: OLPEDGE-1164, OLPEDGE-1165

Signed-off-by: Mykhailo Kuchma <ext-mykhailo.kuchma@here.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,28 @@
+## v1.0.0 (02/12/2019)
+
+**Common**
+
+* Added a new CMake flag `OLP_SDK_BOOST_THROW_EXCEPTION_EXTERNAL` for a case when SDK needs to compile without exceptions, and the user will provide a custom `boost::throw_exception` function.
```
Comment (andescu) [inline CHANGELOG.md]:
```suggestion
* New CMake flag `OLP_SDK_BOOST_THROW_EXCEPTION_EXTERNAL` introduced, to be used when SDK needs to compile without exceptions. The user needs to provide a custom `boost::throw_exception` function.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,28 @@
...
+* `olp::client::HttpResponse` now holds a `std::stringstream` instead of `std::string`, this removes expensive copies.
+* CMake minimum version was changed to 3.9.
```
Comment (andescu) [inline CHANGELOG.md]:
```suggestion
* CMake minimum version increased to 3.9.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,28 @@
...
+* Method `cancel` in CancellationToken is deprecated in favor of a new `Cancel` method.
```
Comment (andescu) [inline CHANGELOG.md]:
```suggestion
* Method `cancel` in CancellationToken is deprecated in favor of the newly introduced `Cancel` method.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,28 @@
...
+* Network logs were improved.
```
Comment (andescu) [inline CHANGELOG.md]:
```suggestion
* Network logs improved for all platforms.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,28 @@
...
+* Various bug fixes and improvements.
+* Network limit was implemented for all platforms. When the limit reached `Send` method will return `NETWORK_OVERLOAD_ERROR`.
```
Comment (andescu) [inline CHANGELOG.md]:
```suggestion
* Network limit now implemented for all platforms. When the limit was reached `Network::Send` method will return `olp::http::ErrorCode::NETWORK_OVERLOAD_ERROR` and client users will receive `olp::client::ErrorCode::SlowDown` as a response.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,28 @@
...
+* Android network client can now be used when Java VM embedded in a native application.
```
Comment (andescu) [inline CHANGELOG.md]:
```suggestion
* Android network client can now be used when Java VM is embedded in a native application.
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,28 @@
...
+
+**olp-cpp-sdk-dataservice-read**
+
+* `VersionedLayerClient` and `VolatileLayerClient` are now movable and non-copyable.
+* `TaskContext` and `PendingRequests` classed moved to core component, and are now public.
```
Comment (andescu) [inline CHANGELOG.md]:
```suggestion
* `TaskContext` and `PendingRequests` classes moved to core component, and are now public.
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,28 @@
...
+* Methods `GetCatalogMetadataVersion` in `CatalogClient` are deprecated, `GetLatestVersion` should be used instead.
```
Comment (andescu) [inline CHANGELOG.md]:
```suggestion
* Method `GetCatalogMetadataVersion` in `CatalogClient` is deprecated, `GetLatestVersion` should be used instead.
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,28 @@
...
+* Deprecated methods were removed.
```
Comment (andescu) [inline CHANGELOG.md]:
```suggestion
* All deprecated methods marked in v0.8.0 are now removed.
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,28 @@
...
+* A new parameter version was added to `PrefetchTileRequest`.
```
Comment (andescu) [inline CHANGELOG.md]:
```suggestion
* A new optional parameter for catalog version added to `PrefetchTileRequest` to be used by `VersionedLayerClient::PrefetchTiles()`. If not provided the latest catalog version will be queried from OLP instead.
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,28 @@
...
+
+**olp-cpp-sdk-dataservice-write**
+
+* Added missing `DATASERVICE_WRITE_API` for various classes.
```
Comment (andescu) [inline CHANGELOG.md]:
```suggestion
* Missing `DATASERVICE_WRITE_API` added to various classes.
```

File: CHANGELOG.md (hunk 10 above)
Comment (HalynaDumych) [reply CHANGELOG.md]: If not provided, the latest catalog version is queried from OLP instead.

**Pull request merged**